src/main/java/redis/embedded/RedisInstance.java

```diff
@@ -42,10 +42,11 @@ public synchronized void start() throws IOException {
             addShutdownHook("RedisInstanceCleaner", checkedToRuntime(this::stop));
             if (serrListener != null)
                 newDaemonThread(() -> logStream(process.getErrorStream(), serrListener)).start();
-            awaitServerReady(process, readyPattern, soutListener);
             if (soutListener != null)
                 newDaemonThread(() -> logStream(process.getInputStream(), soutListener)).start();
 
+            awaitServerReady(process, readyPattern, soutListener);
```

Comment:
The `awaitServerReady()` function will read the same standard output stream as the `soutListener` deamon thread that is being started here. Those two threads will then compete for reading of the input which will break the code. 

Also note that `soutListener` is provided as an argument to `awaitServerReady()` and it is being written to by that function. In `RedisInstance` line 58 the `InputStream` is passed to `findMatchInStream()`, in `IO` on line 68 the `soutListener` is provided a copy of the line.

Reply:
I'm surprised the error is shown with your modification, I would think that error is stuck inside standard err. Are you sure you didn't add an `serrListener` and that is where you saw the libsso message?

I agree though with the sentiment. When something goes wrong you have to go digging through the code and find the `soutListener` and `serrListerner` just to be able to find the real problem. Also, the exception message is too cryptic. A new user of the library wouldn't know what a ready pattern is and most of the time when Redis fails to start there is nothing in standard out, and standard err is not shown. What you see instead is this weird message that ends with a colon and then nothing. There have already been several people bumping into the libsso error, I've had people not realising `serrListener` is a thing, I saw a comment the other day where someone simply gave up on the library after getting the ready pattern exception. 

I'm thinking of rewriting this so that you do see the error stream. Maybe some additional logic to check for common stuff and give a better message. I just haven't gotten to it yet.

Reply:
I pushed a new version of the library. Really just a minor change to the error message. But it should be easier to see what goes wrong because it now includes the error stream output as well in the exception. Fingers-crossed this helps people with what they need to debug their issue.

Reply:
Thank you !